src/components/TextPicker/TextSelectorModal.tsx

```diff
@@ -0,0 +1,62 @@
+import React, {useState} from 'react';
+import {View} from 'react-native';
+import Button from '@components/Button';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import Modal from '@components/Modal';
+import ScreenWrapper from '@components/ScreenWrapper';
+import ScrollView from '@components/ScrollView';
+import TextInput from '@components/TextInput';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import CONST from '@src/CONST';
+import type {TextSelectorModalProps} from './types';
+
+function TextSelectorModal({value, description = '', onValueSelected, isVisible, onClose, ...rest}: TextSelectorModalProps) {
+    const {translate} = useLocalize();
+    const styles = useThemeStyles();
+
+    const [currentValue, setValue] = useState(value);
+
+    return (
+        <Modal
+            type={CONST.MODAL.MODAL_TYPE.RIGHT_DOCKED}
+            isVisible={isVisible}
+            onClose={onClose}
+            onModalHide={onClose}
+            hideModalContentWhileAnimating
+            useNativeDriver
+        >
+            <ScreenWrapper
+                includePaddingTop={false}
+                includeSafeAreaPaddingBottom={false}
+                testID={TextSelectorModal.displayName}
```

Comment:
On safari, the Save button doesn't appear when keyboard is opened

https://github.com/Expensify/App/assets/141406735/723603b8-7f75-4d61-ba97-e092def8e96c

@kosmydel we should add shouldEnableMaxHeight props.